content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -0,0 +1,1387 @@
+- **scaling** is an input field for a Vectror3 value to be used to set the scale of each instance. This property is only visible if there is no wire connected to the scaling input.
+
+#### Advanced
+- **Evaluate context** is used for optimization in the graph when using successive iterable nodes. If this property is set to true, each iteration in this node will also calculate all looped context feeding this node. If there are several iterable nodes connected in sequence, the calculations become exponential with every iteration of the current node calculating every iteration of the previous node back to the beginning of the chain. This can cause a spike in process time in the graph. Disabling evaluate context will prevent nodes from initiating evaluations of previous context each iteration which will greatly speed up the graph. However, since there may be a case where it is desireable to re-evaluate context with each iteration of a node, this parameter is available to the user and defaults to true. If a graph seems to be taking too long to process, check iterable nodes to see if disabling the evaluation of context will help improve performance.
+
+#### Inputs
+- **geometry** is connected to a node providing a source geometry to distribute instances upon.
+- **instance** is connected to a node providing a geometry from which to create instances.
+- **density** is connected to a node providing an Float value in a range of 0.0 - 1.0 which determines the percentage of randomly selected vertices that are eligible for an instance.
+- **matrix** is connected to a node providing a transformation matrix to set position, rotation, and scaling for each instance.
+- **rotation** is connected to a node providing a Vector3 using radians to set the rotation for each instance.
+- **scaling** is connected to a node providing a Vector3 to set the scaling for each instance.
+
+#### Outputs
+- **output** is the geometry generated from the node. No matter how many instances were created with this node, the output will be a single geometry that has been automatically merged. 
+
+## Misc
+These nodes are largely to help with organization and debugging a graph, but there is also some ulility to be found in them. Node graphs are notorious for being difficult to read once they get complex due to the spaghetti of wires that often occurs within large graphs. That's where both the teleport nodes and the elbow node come in. These nodes help reduce some of the wire complexity in graphs, making them much easier to read. Debugging the graph can also be challenging especially when large arrays of values are being pushed through the graph. The debug node is crucial for being able to see values passed along any wire at any point. Lastly, the GeometryInfo node can be used as easily to debug the graph as it can to drive logic within.

```

Comment:
nit: ulility => utility

Reply:
fixed in https://github.com/BabylonJS/Documentation/pull/963/commits/5395b92d4e3c0f28c82faf008026b09b521801ce